tests/functional/z_cluster/cluster_expansion/test_resize_osd.py

```diff
@@ -216,3 +226,44 @@ def test_resize_osd_with_resource_delete(self, resource_name, num_of_iterations)
         self.new_storage_size = basic_resize_osd(self.old_storage_size)
         delete_resource_multiple_times(resource_name, num_of_iterations)
         self.verification_steps_post_resize_osd()
+
+    @tier4b
+    @polarion_id("OCS-5785")
+    def test_resize_osd_when_capacity_near_full(
+        self, benchmark_workload_storageutilization
+    ):
+        """
+        Test resize OSD when the cluster capacity is near full
+
+        """
+        target_percentage = 70
+        logger.info(
+            f"Fill up the cluster to {target_percentage}% of it's storage capacity"
+        )
+        benchmark_workload_storageutilization(target_percentage=70)
+        self.prepare_data_before_resize_osd()
+        self.new_storage_size = basic_resize_osd(self.old_storage_size)
+        self.verification_steps_post_resize_osd()
+
+    @tier4a
+    @pytest.mark.last
+    @polarion_id("OCS-5786")
+    def test_resize_osd_for_large_diff(self):
+        """
+        Test resize osd for large differences. The test will increase the osd size to 4Ti.
+        If the current OSD size is less than 1024Gi, we will skip the test, as the purpose of the test
+        is to check resizing the osd for large differences.
+
+        """
+        logger.info(f"The current osd size is {self.old_storage_size}")
+        current_osd_size_in_gb = convert_device_size(self.old_storage_size, "GB", 1024)
+        max_osd_size_in_gb = 1024
+        if current_osd_size_in_gb > max_osd_size_in_gb:
+            pytest.skip(
+                f"The test will not run when the osd size is less than {max_osd_size_in_gb}Gi"
+            )
+
+        self.prepare_data_before_resize_osd()
+        self.new_storage_size = "4Ti"
```

Comment:
4Ti is constant? Do we have enough resources for it?

Reply:
It's one of the test cases: https://docs.google.com/spreadsheets/d/1wCSwMk_L4SikzZ2UarUH1-VCXyGkHs-ZNdDEtWPG7f4/edit#gid=105186681&range=B14. Maybe we can add a condition at setup to skip the test if we have more than 3 OSDs? What do you think should be the maximum resource limits?